src/test/java/TestSuite.java

```diff
@@ -0,0 +1,199 @@
+import com.google.gson.JsonSyntaxException;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.junit.platform.suite.api.IncludeTags;
+import org.junit.platform.suite.api.SelectClasses;
+import org.junit.platform.suite.api.Suite;
+import parser.JsonParser;
+import parser.NoSuchFileException;
+import parser.Parser;
+import shop.Cart;
+import shop.RealItem;
+import shop.VirtualItem;
+
+import java.io.File;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+@Suite
+@SelectClasses({JsonParserTest.class, RealItemTest.class, VirtualItemTest.class, CartTest.class})
+@IncludeTags({"JsonParserTest", "RealItemTest", "VirtualItemTest", "CartTest"})
+class TestSuite {
+}
+
+@Tag("JsonParserTest")
+@DisplayName("JsonParser Class Tests")
+class JsonParserTest {
+    Parser parser;
+    File file;
+    Cart cart;
+
+    @BeforeEach
+    public void setup() {
+        parser = new JsonParser();
+        cart = new Cart("test-cart");
+        file = new File("src/main/resources/" + cart.getCartName() + ".json");
+    }
+
+    @AfterEach
+    public void cleanUp() {
+        file.deleteOnExit();
+        cart = null;
+    }
+
+    @DisplayName("Should Write To File")
+    @Test
+    public void shouldWriteToFile() {
+        parser.writeToFile(cart);
+        // checking that file is not empty
+        assertNotEquals(0, file.length());
+    }
+
+    @DisplayName("Should Read From File")
+    @ParameterizedTest
+    @ValueSource(strings = {"andrew-cart", "eugen-cart"})
+    public void shouldReadFromFile(String cartName) {
+        cart = parser.readFromFile(new File("src/main/resources/" + cartName + ".json"));
```

Comment:
Please don`t use global variable, as it already had initialized. 
 Cart cart = parser.readFromFile(new File("src/main/resources/" + cartName + ".json"));
And it is require to write sth into file first